src/image.rs

```diff
@@ -20,30 +20,38 @@ pub fn draw_kind(
     canvas: &mut Canvas,
 ) {
     match kind {
+        usvg::ImageKind::SVG(ref subtree) => {
+            draw_svg(subtree, view_box, canvas);
+        }
+        #[cfg(feature = "image")]
         usvg::ImageKind::JPEG(ref data) => {
             match read_jpeg(data) {
                 Some(image) => { draw_raster(&image, view_box, rendering_mode, canvas); }
                 None => log::warn!("Failed to decode a JPEG image."),
             }
         }
+        #[cfg(feature = "image")]
         usvg::ImageKind::PNG(ref data) => {
             match read_png(data) {
                 Some(image) => { draw_raster(&image, view_box, rendering_mode, canvas); }
                 None => log::warn!("Failed to decode a PNG image."),
             }
         }
+        #[cfg(feature = "image")]
         usvg::ImageKind::GIF(ref data) => {
             match read_gif(data) {
                 Some(image) => { draw_raster(&image, view_box, rendering_mode, canvas); }
                 None => log::warn!("Failed to decode a GIF image."),
             }
         }
-        usvg::ImageKind::SVG(ref subtree) => {
-            draw_svg(subtree, view_box, canvas);
+        #[cfg(not(feature = "image"))]
+        _ => {
+            log::debug!("Not decoding embedded image because feature is not enabled.");
         }
     }
 }
 
+#[cfg(feature = "image")]
```

Comment:
I think we could move all raster methods and types into a `raster_image` submodule. I.e. `mod raster_image {}`. This would introduce nesting, but would make our intentions clearer.